src/ServiceFabricPersistence/SynchronizedStorage/ServiceFabricStorageSession.cs

```diff
@@ -0,0 +1,64 @@
+namespace NServiceBus.Persistence.ServiceFabric
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using Microsoft.ServiceFabric.Data;
+    using Outbox;
+    using Transport;
+
+    class ServiceFabricStorageSession : ICompletableSynchronizedStorageSession, IServiceFabricStorageSession
+    {
+        public ServiceFabricStorageSession(IReliableStateManager stateManager, TimeSpan transactionTimeout)
+        {
+            TransactionTimeout = transactionTimeout;
+            StateManager = stateManager;
+        }
+
+        public IReliableStateManager StateManager { get; }
+
+        public ITransaction Transaction { get; private set; }
+
+        public TimeSpan TransactionTimeout { get; }
+
+        public void Dispose()
+        {
+            if (!disposed && ownsTransaction)
+            {
+                Transaction.Dispose();
+                disposed = true;
+            }
+        }
+
+        public ValueTask<bool> TryOpen(IOutboxTransaction transaction, ContextBag context,
+            CancellationToken cancellationToken = default)
+        {
+            if (transaction is ServiceFabricOutboxTransaction outboxTransaction)
+            {
+                Transaction = outboxTransaction.Transaction;
```

Comment:
i'm assuming the context is irrelevant here since you're not capturing it? idem below

Reply:
Nvm i can see it's irrelevant in this case